File: binary_search/35. Search Insert Position/35.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    let l = -1
    let r = nums.length
    while (r - l > 1) {
        let m = Math.floor( (r + l) / 2 )
        if (nums[m] === target) return m
        if (nums[m] > target) r = m
        else l = m
    } 
    return r
};

File: binary_search/35. Search Insert Position/35.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l = -1
        r = len(nums)
        while ((r - l ) > 1):
            m = ( r + l ) // 2
            if nums[m] == target: return m
            if nums[m] > target: r = m
            else: l = m
        
        return r